File: wrapper/ios/Feature/ZeticMLangeModelWrapper.swift
import Foundation
import ZeticMLange

class ZeticMLangeModelWrapper: PipelineInferenceModel {
    private let model: ZeticMLangeModel
    
    init(_ tokenKey: String, _ modelKey: String) {
        self.model = (try? ZeticMLangeModel(tokenKey, modelKey))!
    }
    
    func inference(_ input: [Data]) -> [Data] {
        do {
            try model.run(input)
            return model.getOutputDataArray()
        } catch {
            return []
        }
    }
}

File: wrapper/android/ext/src/main/java/com/zeticai/mlange/feature/ZeticMLangeModelWrapper.kt
package com.zeticai.mlange.feature

import android.content.Context
import com.zeticai.mlange.core.model.ZeticMLangeModel
import com.zeticai.mlange.core.tensor.Tensor

class ZeticMLangeModelWrapper(
    context: Context,
    personalKey: String,
    modelKey: String
) : PipelineInferenceModel {
    private val model = ZeticMLangeModel(context, personalKey, modelKey)

    override fun inference(inputs: Array<Tensor>): Array<Tensor> {
        return model.run(inputs)
    }
}
